MNT co-vendor array-api-{compat, extra} (#30340)
scikit-learn/scikit-learn: doc/modules/array_api.rst

The `Array API <https://data-apis.org/array-api/latest/>`_ specification defines
a standard API for all array manipulation libraries with a NumPy-like API.

> Scikit-learn vendors pinned copies of
> `array-api-compat <https://github.com/data-apis/array-api-compat>`__
> and `array-api-extra <https://github.com/data-apis/array-api-extra>`__.

Author: ```suggestion
Scikit-learn vendors pinned copies of
`array-api-compat <https://github.com/data-apis/array-api-compat>`__
and `array-api-extra <https://github.com/data-apis/array-api-extra>`__.

Scikit-learn's support for the array API standard requires the environment variable
`SCIPY_ARRAY_API` to be set to `1` before importing `scipy` and `scikit-learn`:

.. prompt:: bash $

   export SCIPY_ARRAY_API=1

Please note that this environment variable is intended for temporary use.
For more details, refer to SciPy's `Array API documentation
<https://docs.scipy.org/doc/scipy/dev/api-dev/array_api.html#using-array-api-standard-support>`_.
```

common tests to verify that the estimators' results are the same when using
vanilla NumPy and Array API inputs.

> To run the full set of checks you need to install both
> `PyTorch <https://pytorch.org/>`_ and `CuPy <https://cupy.dev/>`_ and have
> a GPU. Checks that can not be executed or have missing dependencies will be

Reviewer: This is half suggestion and half request for comment (ill make a new PR):

```suggestion
To run the full set of checks you need to install 
`array-api-strict <https://data-apis.org/array-api-strict/>`_ which does not require a GPU. If you
have a GPU you can install `PyTorch <https://pytorch.org/>`_ or `CuPy <https://cupy.dev/>`_ instead.
Checks that can not be executed or have missing dependencies will be
```

I should have updated the docs in #30090

Reviewer: Let's discuss/wordsmith in a new PR. At least for me this is not important enough to delay merging this.

I'll make a PR
